### PR TITLE
[core] Add bindings for custom menus

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by zach2good** [comment]
image

File: scripts/commands/menu.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+-----------------------------------
+
+cmdprops =
+{
+    permission = 1,
+    parameters = ""
+}
+
+function onTrigger(player)
+    player:customMenu({ "Test Menu (Play Effect)",
```
Comment (claywar) [inline scripts/commands/menu.lua]: This styling kills me, and I'd hate to set the example for further users.  Perhaps define the table outside of the function call with our style guides in mind?

Comment (zach2good) [reply scripts/commands/menu.lua]: Happy to accept other stylings, all Allman?

Comment (TeoTwawki) [reply scripts/commands/menu.lua]: I think we sort of cover this case:

https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md#inline-tables

Comment (zach2good) [reply scripts/commands/menu.lua]: I think I'd be OK adjusting this rule to be "define your table beforehand and pass it in", I never really liked this edge case

**Review by claywar — APPROVED** [review]

**Pull request merged**